### PR TITLE
Remove offer duplicates in spread view

### LINKED ISSUE #1202: Offer counts in Market->Spread tab temporarily doubles
Here are two screenshots, one taken right after the other.

screen shot 2018-01-16 at 3 05 00 pm

image

Note how the offer counts in the first are exactly double the counts in the second.

The only thing I did in between these two screenshots was click on "Buy BTC" and "Sell BTC" in order to examine whether there was a single market maker responsible for many of these trades. I noticed immediately on those screens that the numbers were off (there weren't ~180 trades there), and when I came back to the Spread tab, I saw the view in the second screenshot.

Whatever is causing this behavior is probably what was going on when I tweeted this the other day: https://twitter.com/bisq_network/status/952485704994639872. So unfortunately, that was probably a false "record" being set.

### PR DESCRIPTION
**Pull request by ripcurlx**
Fixes #1202.

I was only able to reproduce it in a test in the SpreadModel so I added an additional safeguard there.

### CODE REVIEW

File: src/main/java/bisq/desktop/main/market/spread/SpreadViewModel.java [hunk 1]
```diff
@@ -90,6 +94,12 @@ protected void deactivate() {
         offerBookListItems.removeListener(listChangeListener);
     }
 
+    private static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor)
+    {
```
Comment (cbeams) [inline src/main/java/bisq/desktop/main/market/spread/SpreadViewModel.java]: Nit: place opening brace on same line

File: src/test/java/bisq/desktop/main/market/spread/SpreadViewModelTest.java [hunk 2]
```diff
@@ -58,14 +62,38 @@ public void testMaxCharactersForAmountWithNoOffers() {
...
-        offerBookListItems.addAll(make(btcItem.but(with(OfferBookListItemMaker.amount, 1403000000L))));
+        offerBookListItems.addAll(make(btcBuyItem.but(with(OfferBookListItemMaker.amount, 1403000000L))));
         assertEquals(7, model.maxPlacesForAmount.intValue()); //14.0300
     }
+
+    @Test
+    public void testfilterSpreadItemsForUniqueOffers() {
+        OfferBook offerBook = mock(OfferBook.class);
+        PriceFeedService priceFeedService = mock(PriceFeedService.class);
+        final ObservableList<OfferBookListItem> offerBookListItems = FXCollections.observableArrayList();
```
Comment (cbeams) [inline src/test/java/bisq/desktop/main/market/spread/SpreadViewModelTest.java]: Nit: don't use `final` modifier on local variables unless strictly necessary (same for line 85 below.

**Review by cbeams — APPROVED** [review]
utACK. Great that you were able to repro this at all, @ripcurlx! You'll see I called out a couple style nits, but looks good otherwise. Feel free to merge.

File: src/test/java/bisq/desktop/main/market/spread/SpreadViewModelTest.java (hunk 2 above)
Comment (ripcurlx) [reply src/test/java/bisq/desktop/main/market/spread/SpreadViewModelTest.java]: I use it out of a habit. Does it have a disadvantage to use final everywhere by default and only not use it where necessary?

**Pull request merged**